roles/ci_local_storage/tasks/cleanup.yml

```diff
@@ -44,29 +44,41 @@
     name: "{{ item }}"
   loop: "{{ cifmw_cls_pvs }}"
 
-- name: Get the worker nodes
+- name: Get k8s nodes
+  when: cifmw_openshift_kubeconfig is defined
   kubernetes.core.k8s_info:
     kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
     api_key: "{{ cifmw_openshift_token | default(omit)}}"
     context: "{{ cifmw_openshift_context | default(omit)}}"
     kind: Node
-    label_selectors:
-      - node-role.kubernetes.io/worker
-  register: _node_info
+  register: cifmw_ci_local_storage_k8s_nodes_out
 
-- name: Get all the worker nodes
-  ansible.builtin.set_fact:
-    cifmw_cls_nodes: >-
-      {{ _node_info.resources |
+- name: Set the hosts k8s ansible hosts
+  vars:
+    k8s_nodes_hostnames: >-
+      {{
+          cifmw_ci_local_storage_k8s_nodes_out.resources |
+          default([]) |
           selectattr("metadata.name", "defined") |
           map(attribute="metadata.name")
       }}
+  ansible.builtin.set_fact:
+    cifmw_ci_local_storage_k8s_hosts: >-
+      {{
+          hostvars | dict2items |
+          selectattr("value.ansible_hostname", "defined") |
+          selectattr("value.ansible_hostname", "in", k8s_nodes_hostnames) |
+          map(attribute="key") |
+          list
+      }}
 
-- name: Delete directories on worker nodes
+- name: Create directories on worker node
```

Comment:
Small nit, this should be `Delete directories on worker node`

Reply:
Good one.